src/lib.rs

```diff
@@ -431,11 +450,20 @@ impl JayLink {
             caps: Capabilities::from_raw_legacy(0), // dummy value
             interface: Interface::Spi,              // dummy value, must not be JTAG
             interfaces: Interfaces::from_bits_warn(0), // dummy value
+            conn: None,
             handle,
         };
         this.fill_capabilities()?;
         this.fill_interfaces()?;
 
+        let mut conn = Connection::default();
+        if this.caps.contains(Capability::Register) {
+            debug!("Registering connection");
+            this.register(&mut conn)?;
```

Comment:
What happens if the device runs out of connections? Is that something that can happen? Will it reject new connections from being registered or will older ones be dropped automatically?

(this is something that could easily happen when an app either calls `process::exit` or is killed via Ctrl+C)

Reply:
On my old-clone-firmware-whatever device, it never runs out of connections. It always has four, and newer ones just overwrite older ones, it's like a 4-size ring buffer.

Reply:
Okay, good. I was going to test this on my (original) J-Link but noticed that the `list` example is failing with `<USB error while reading from device: Overflow>`, so it looks like this doesn't work anymore with some hardware? (this is a J-Link Base Compact version 10.1)

Reply:
hm. Where is it failing? Is that after receiving the reply for registration? If so, can you dump the lengths (after `// Receive connection table`) and/or the whole received `buf`?

Reply:
It's just the read of 76 Bytes failing:

```
[2021-07-25T12:15:04Z DEBUG jaylink] Registering connection
[2021-07-25T12:15:04Z TRACE jaylink] write 14 bytes: [9, 64, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
[2021-07-25T12:15:04Z TRACE jaylink] reading 76 bytes
Bus 001 Address 029 Port 003: VID=1366 PID=0105 – <USB error while reading from device: Overflow>
```

Reply:
huh, so it is an actual USB level error? hmm. Does OpenOCD work with this device?

Reply:
Yes, it should work. I believe we need to change how reads/writes work, because libjaylink seems to use an extra buffer, while we directly call into libusb (which will refuse to read more than 76 Bytes into the 76 Byte buffer). You might not be hitting this problem because the clone firmware always returns a fixed-length response here.

Reply:
Yes, the FW I have always returns 76. Hmm. I guess we could use an arbitrary big buffer just for this? ¯\\\_(ツ)\_/¯